### DISCUSSION TITLE
Building codium on NetBSD

### DISCUSSION THREAD

**Discussion by ghbpeter** [VSCodium/vscodium#2175] [Q&A]
Hello,
I am trying to build codium on NetBSD 10.0. After unpacking 1.96.2 and running build/build.sh command prompt outputs:
```
...
+++ ../update_settings.sh
300s/'default': true/'default': false/
sed: -E: No such file or directory
219s/'default': TelemetryConfiguration.ON/'default': TelemetryConfiguration.OFF/
sed: -E: No such file or directory
114s/'default': true/'default': false/
sed: -E: No such file or directory
#

```
and I can find file vscode/cli/src/bin/code/main.rs

I am C++ developer and I don't know this script world. Is main.rs final file from build ? Or build was aborted due to above errors ?
I don't need to create installation package, sufficient should be running codium from build. 
What should I do to continue creating executable binary ... /bin/codium ?

**Comment by Jeong-ilKajokaya** [comment]
Hi! I can help clarify what’s going on.

The errors you’re seeing:

```
sed: -E: No such file or directory
```

are **because NetBSD’s `sed` doesn’t support `-E` for extended regex**. On Linux/macOS, `-E` works fine, but on BSD systems you usually need to use `-r` or adjust the `sed` syntax. This is why those substitutions in `update_settings.sh` are failing.

---

### About `main.rs`

* `vscode/cli/src/bin/code/main.rs` is **the source code** for the CLI binary.
* It is **not the final executable**. Rust source files need to be **compiled** with `cargo` (Rust’s build system) to produce the executable.
* So your build did **not complete successfully** — the script errors mean some replacements and setup didn’t run, and the binary `/bin/codium` was likely not created yet.

---

### What you can do

1. **Install GNU `sed`** on NetBSD, which supports `-E`. Often this can be done via `pkgin` or `pkg_add`. For example:

```sh
pkgin install gsed
```

Then modify the script to call `gsed` instead of `sed`, or update your PATH:

```sh
export PATH="/usr/pkg/bin:$PATH"
```

2. **Verify Rust and Cargo are installed**:

```sh
rustc --version
cargo --version
```

3. **Run the build again**:

```sh
./build/build.sh
```

4. If you only need the executable **without creating an installation package**, after a successful build you should find the binary in the `out` folder or in `./VSCode-linux-x64` (depending on your build configuration).

---

**Summary:**
The build was **not completed** because `sed -E` failed. `main.rs` is just source code, not the final binary. Installing a compatible `sed` and re-running the build should allow the compilation to finish and create `/bin/codium`.

---

If you want, I can also give a **NetBSD-ready version of `update_settings.sh`** that avoids `-E` so it works without installing GNU sed.
